test/cql-pytest/test_tools.py

```diff
@@ -146,6 +136,7 @@ def scylla_sstable(table_factory, cql, ks, data_dir):
 
     try:
         with nodetool.no_autocompaction_context(cql, f"{ks}.{table}"):
+            nodetool.flush(cql, f"{ks}.{table}")
```

Comment:
Note that all table factories also do intermediate flushes, not just a final flush. They are all designed to produce a certain number of sstables. Taking this control away from them, even if just partially, is weakening the test.

If you want to test a new kind of flush, I suggest to write a new test for it, instead of interfering with existing ones.

Reply:
dropped all patches test test_tools.py from the series (we can rename it in a followup PR)